clients/graphql-kotlin-client/src/main/kotlin/com/expediagroup/graphql/client/types/AutomaticPersistedQueriesSettings.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.client.types
+
+data class AutomaticPersistedQueriesSettings(
```

Comment:
APQ requests can also be send through HTTP POST, this implementation forces GET, maybe in the APQSetting we can specify which HTTP Verb the client wish to use.

Reply:
@samuelAndalon 
I thought APQ had to be GET.
[The Apollo documentation](https://www.apollographql.com/docs/apollo-server/performance/apq/
) was also written in GET.

For a POST request in GraphQL, [the query field in the body should be required](https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md).
If APQ allows POST requests, then inevitably a large query string will be included in the request, I think which is inconsistent with the benefits of APQ.

Am I correct that the query field is optional when APQ is used in the POST method?
If so, I will change `interface GraphQLClientRequest.query` to nullable.
Also can you please share any detailed documentation on APQ specs (especially HTTP request and response)?

Reply:
for APQ requests, the `query` field of the JSON payload is not required. 
graphql kotlin server allows APQ requests for both GET and POST, i think we should be consistent with the client.

Reply:
>If APQ allows POST requests, then inevitably a large query string will be included in the request, I think which is inconsistent with the benefits of APQ.

APQ worst case scenario (when persited query is not found) needs to send a second request, with persitedQuery extension and query

Reply:
https://opensource.expediagroup.com/graphql-kotlin/docs/server/automatic-persisted-queries/

Reply:
AFAIK there is no official APQ spec yet and there are number of implementations out there (including the linked one from Apollo). Per the graphql http spec [discussion](https://github.com/graphql/graphql-over-http/issues/216) and [issue](https://github.com/graphql/graphql-over-http/issues/38) it is still TBD.

As for changing the `query` to be nullable -> I have no idea what sort of impact that would be, guessing it should be ok as it is a handled by the client internals.

Reply:
The server logic itself should handle the nullable `query`, at least graphql-kotlin server does that.

if you dont send an APQ request (request without entension), and no query, then the server will return 400.

Reply:
when added APQ in graphql-kotlin had to make query value default to ""

https://github.com/ExpediaGroup/graphql-kotlin/commit/f773327b126832c6176c0fd64e26aa6fea023d94#diff-b4c1162353124139fdbd7ef3c7f304d95203c1c7b8392ed9a153ed7a4c67f96eR42

Reply:
@samuelAndalon @dariuszkuc 
Thank you for your comment!
I've change to allow POST requests and `GraphQLClientRequest.query` to be nullable.